torch_xla/experimental/apply_layers.py

```diff
@@ -0,0 +1,126 @@
+from typing import Iterable, Mapping, Sequence, Dict, Tuple
+
+import torch
+import torch.nn as nn
+from torch.utils._pytree import tree_map
+from functorch.compile import default_partition
+
+import torch_xla
+from torch_xla.experimental.scan import scan
+
+
+def apply_layers(layers: Iterable[torch.nn.Module],
```

Comment:
the name `apply_layers` does not really reflect that we are trying to pack the layers into a while loop, is this name coming from JAX?

Reply:
Nope. I invented it.

Reply:
Not sure what's a better name. In flax (the neural network library on top of jax), it's also called `scan`, but it's in a different names space: `flax.nn.scan` vs `jax.lax.scan`. The flax scan handles modules and the jax scan handles tensors.

Reply:
Ok, I asked an LLM for ideas and I think we can go with `scan_layers`: it's a version of `scan` that handles nn layers